akka-remote/src/main/scala/akka/remote/artery/compress/CompressionTable.scala

```diff
@@ -7,41 +7,52 @@ package akka.remote.artery.compress
 import java.util
 import java.util.Comparator
 
+import org.agrona.collections.Object2IntHashMap
+
 /**
  * INTERNAL API: Versioned compression table to be advertised between systems
  *
  * @param version Either -1 for disabled or a version between 0 and 127
  */
-private[remote] final case class CompressionTable[T](originUid: Long, version: Byte, dictionary: Map[T, Int]) {
-  import CompressionTable.NotCompressedId
+private[remote] final class CompressionTable[T](
+    val originUid: Long,
+    val version: Byte,
+    _dictionary: Object2IntHashMap[T]) {
+
+  def dictionary: Map[T, Int] = {
```

Comment:
This is used from deserialization of CompressionTable, but that is for the periodic advertisements so not very often. Same for the CompressionTable.apply.

Reply:
And also `invert` :+1:

Reply:
yes, invert is probably very slow, and we have some todo there, but not on hot path